File: dcp-solutions/2024-10-05/solution.go
```markdown
package main

import "container/list"

type ListNode struct {
	Val int
	Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	dummy := &ListNode{-1, head}
	first, second := dummy, dummy
	for i := 0; i < n+1; i++ {
		first = first.Next
	}
	for first != nil {
		first = first.Next
		second = second.Next
	}
	second.Next = second.Next.Next
	return dummy.Next
}
```

File: dcp-solutions/2024-10-05/solution.py
```python
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def removeNthFromEnd(head, n):
    # if list has only one node
    if not head or not head.next:
        return None
    
    # two pointers, first pointer is always ahead of the second by k nodes
    lead = head
    tail = head
    for i in range(n + 1):
        if not lead.next:
            break
        lead = lead.next
        
    # remove nth from end node
    if not lead.next:  
        return None
    else:
        while lead.next.next:
            lead = lead.next 
            tail = tail.next    
        tail.next = tail.next.next   
    return head
```